poc_iot_injector/src/cli/generate.rs

```diff
@@ -33,33 +38,45 @@ impl Cmd {
         let poc_oracle_key = settings.keypair()?;
         let shared_key = Arc::new(poc_oracle_key);
 
+        let success_counter = Arc::new(Mutex::new(0));
+        let failure_counter = Arc::new(Mutex::new(0));
+
         store
             .source_unordered(LOADER_WORKERS, stream::iter(file_list).map(Ok).boxed())
-            .for_each_concurrent(STORE_WORKERS, |msg| {
+            .try_for_each_concurrent(STORE_WORKERS, |msg| {
                 let shared_key_clone = shared_key.clone();
+                let success_counter_ref = Arc::clone(&success_counter);
+                let failure_counter_ref = Arc::clone(&failure_counter);
                 async move {
-                    match msg {
-                        Ok(m) => process_msg(m, shared_key_clone, before_ts).await,
-                        Err(e) => tracing::error!("unable to process msg due to: {:?}", e),
+                    if process_msg(msg, shared_key_clone, before_ts).await.is_ok() {
+                        *success_counter_ref.lock().unwrap() += 1;
+                    } else {
+                        *failure_counter_ref.lock().unwrap() += 1;
                     }
+                    Ok(())
                 }
             })
-            .await;
+            .await?;
+
+        tracing::debug!("success_counter: {:?}", success_counter.lock().unwrap());
+        tracing::debug!("failure_counter: {:?}", failure_counter.lock().unwrap());
```

Comment:
```suggestion
        tracing::debug!("success count: {:?}; failure count: {:?}", success_counter.lock().unwrap(), failure_counter.lock().unwrap());
```
are these unwraps more or less guaranteed safe operations since we're just handling the handoff of a mutex lock and trusting the compiler to negotiate that properly or do we just not care as much since it's a cli and not a production running server process?

Reply:
Yeah pretty much, I just added the counters for inspection. The `generate` cli isn't meant for prod.